python/foglamp/services/core/api/service.py

```diff
@@ -171,14 +173,21 @@ def get_service_records():
         try:
             # Create a configuration category from the configuration defined in the plugin
             category_desc = plugin_config['plugin']['description']
-            config_mgr = ConfigurationManager(storage)
             await config_mgr.create_category(category_name=name,
                                              category_description=category_desc,
                                              category_value=plugin_config,
                                              keep_original_items=True)
             # Create the parent category for all South services
             await config_mgr.create_category("South", {}, "South microservices", True)
             await config_mgr.create_child_category("South", [name])
+
+            # If config is in POST data, then update the value for each config item
+            if config is not None:
+                if not isinstance(config, dict):
+                    raise ValueError('Config must be a JSON object')
+                for k, v in config.items():
```

Comment:
Personally, I wanted this to be done in a single call instead of a loop. But to avoid overwriting default, this is okay!